Neos.Neos/Classes/Fusion/MenuItem.php

```diff
@@ -11,58 +11,17 @@
  */
 final class MenuItem
 {
-    protected Node $node;
-
-    protected ?MenuItemState $state;
-
-    protected ?string $label;
-
-    protected int $menuLevel;
-
-    /**
-     * @var array<int,MenuItem>
-     */
-    protected array $children;
-
-    protected ?string $uri;
-
     /**
      * @param array<int,MenuItem> $children
      */
     public function __construct(
-        Node $node,
-        ?MenuItemState $state = null,
-        ?string $label = null,
-        int $menuLevel = 1,
-        array $children = [],
-        string $uri = null
+        public readonly Node $node,
+        public readonly ?MenuItemState $state = null,
+        public readonly ?string $label = null,
+        public readonly int $menuLevel = 1,
+        public readonly array $children = [],
+        public readonly ?string $uri = null
```

Comment:
nitpick: Like above I would suggest to make the class `readonly` (can't suggest that because it's not part of the changeset)
```suggestion
        public Node $node,
        public ?MenuItemState $state = null,
        public ?string $label = null,
        public int $menuLevel = 1,
        public array $children = [],
        public ?string $uri = null
```

Reply:
Adjusted as suggested